hw4/briansp8210/matrix.h

```diff
@@ -0,0 +1,74 @@
+#ifndef MATRIX_H_
+#define MATRIX_H_
+
+#include <cstddef>
+#include <vector>
+#include <cstdint>
+#include <limits>
+#include <cstdlib>
+
+static uint64_t bytes_allocated;
```

Comment:
I don't think this is a good approach, you can see others' implementations.

Reply:
Is it because the use of global variable?

Reply:
yes. global variables are hard to maintain and scale.